backend/tracim_backend/lib/core/user_custom_properties.py

```diff
@@ -0,0 +1,97 @@
+import typing
+from typing import Any
+from typing import Dict
+
+import jsonschema
+from jsonschema import ValidationError
+
+from tracim_backend.config import CFG
+from tracim_backend.exceptions import TracimValidationFailed
+from tracim_backend.lib.utils.dict_parsing import get_translation_key_from_dict
+from tracim_backend.lib.utils.dict_parsing import translate_dict
+from tracim_backend.lib.utils.translation import TranslationSource
+from tracim_backend.lib.utils.translation import Translator
+from tracim_backend.models.auth import User
+from tracim_backend.models.tracim_session import TracimSession
+from tracim_backend.models.user_custom_properties import UserCustomProperties
+
+# JSONSchema field useful to be translated.
+# note: enumNames is not standard but supported
+# see: https://react-jsonschema-form.readthedocs.io/en/latest/usage/single/#custom-labels-for-enum-fields
+JSON_SCHEMA_KEYS_TO_TRANSLATE = ["enumNames", "title", "description"]
+# INFO - G.M - 2020-01-11 - UISchema field useful to be translated
+# see https://react-jsonschema-form.readthedocs.io/en/latest/api-reference/uiSchema/
+UI_SCHEMA_KEYS_TO_TRANSLATE = ["title", "description", "placeholder", "help"]
+UI_SCHEMA_KEYS_TO_TRANSLATE += ["ui:{}".format(key) for key in UI_SCHEMA_KEYS_TO_TRANSLATE]
+
+
+class UserCustomPropertiesApi:
+    """Api to query user custom properties"""
+
+    def __init__(self, current_user: User, session: TracimSession, app_config: CFG):
+        self._current_user = current_user
+        self._session = session
+        self._config = app_config
+        self._translator = Translator(
+            app_config=self._config,
+            default_lang=self._current_user.lang if self._current_user else None,
+            fallback_lang=self._config.DEFAULT_LANG,
+            default_source=TranslationSource.CUSTOM_PROPERTIES,
+        )
+
+    def get_custom_properties(self) -> UserCustomProperties:
+        query = self._session.query(UserCustomProperties)
+        return query.filter(UserCustomProperties.user_id == self._current_user.user_id).one()
```

Comment:
Technically, not really, it can be discussed but current user given to UserCustomPropertiesApi() in endpoint is candidate one, not current one.
As most access rules remain the role of the api, there is no need to specify the current_user as "current connected one". Note the same usage of current user as candidate user in lib is the same one as for user_config (which i copy paste the code a lot for this feature, because base logic is the same).

Reply:
There is not really a classic way here. So i do think it might be acceptable to do so now.
I do think there is also some really special case in code where we do need 3 different users : 
- the real current user or none (if admin)
- the current user of the api to have similar access to things
- the user id given for this specific request.